controllers/directivebreakdown_controller.go

```diff
@@ -122,30 +129,74 @@ func (r *DirectiveBreakdownReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	argsMap, err := dwdparse.BuildArgsMap(dbd.Spec.Directive)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	commonResourceName, commonResourceNamespace := getStorageReferenceNameFromDBD(dbd)
-	var persistentInstance *dwsv1alpha1.PersistentStorageInstance
 
-	// If the lifetime of the storage is persistent, create PersistentStorageInstance
-	if dbd.Spec.Lifetime == dwsv1alpha1.DirectiveLifetimePersistent {
+	switch argsMap["command"] {
+	case "create_persistent":
 		var result controllerutil.OperationResult
-		result, persistentInstance, err = r.createOrUpdatePersistentStorageInstance(ctx, dbd, commonResourceName, log)
+		result, persistentStorage, err := r.createOrUpdatePersistentStorageInstance(ctx, dbd, commonResourceName, argsMap)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
 
 		// If we failed to create the persistent instance or we created it this pass, requeue.
-		if persistentInstance == nil || result == controllerutil.OperationResultCreated {
-			return ctrl.Result{Requeue: true}, nil
+		if persistentStorage == nil || result == controllerutil.OperationResultCreated {
+			return ctrl.Result{}, nil
 		}
 
-		if persistentInstance.Status.Servers == (v1.ObjectReference{}) {
-			return ctrl.Result{Requeue: true}, nil
+		if persistentStorage.Status.Servers == (v1.ObjectReference{}) {
```

Comment:
Would you please add a comment to ease the burden when I read this again that this is checking whether the Servers is created? 
Also, will this controller wake up when the persistent instance sets Status.Servers because it owns it?